optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -533,10 +567,61 @@ def tune_params(self, target_param_names, n_trials, sampler):
         # Add tuning history.
         self.tuning_history += objective.report
 
-        if self.compare_validation_metrics(study.best_value, self.best_score):
-            self.best_score = study.best_value
+        best_trial = study.best_trial
+
+        # The values of complete trials are not None. The assertion is just for mypy.
+        assert best_trial.value is not None
+
+        if self.compare_validation_metrics(best_trial.value, self.best_score):
+            self.best_score = best_trial.value
             self.best_booster = objective.best_booster
 
-            updated_params = {p: study.best_trial.params[p] for p in target_param_names}
+            updated_params = {p: best_trial.params[p] for p in target_param_names}
             self.lgbm_params.update(updated_params)
             self.best_params.update(updated_params)
+
+    def _create_stepwise_study(
+        self, study: "optuna.study.Study", step_name: str
+    ) -> "optuna.study.Study":
+
+        # This class is assumed to be passed to a sampler and a pruner corresponding to the step.
+        class _StepwiseStudy(optuna.study.Study):
```

Comment:
I'm wondering whether a user wouldn't want to use this class (publicly) to diagnose the study. 🤔

Reply:
On a second thought, maybe it's wiser to keep it private reducing complexity.

Reply:
> reducing complexity

I agree. I feel `_StepwiseStudy` is too specific for most of the users, and want to keep it private like `_BracketStudy` in HyperbandPruner.
https://github.com/optuna/optuna/blob/master/optuna/pruners/hyperband.py#L144